File: test/solutions/11.jl
function maxArea(height::Array{Int64,1})::Int64
    n = length(height)
    left = 1
    right = n
    max_area = 0
    while left < right
        max_area = max(max_area, (right - left) * min(height[left], height[right]))
        if height[left] < height[right]
            left += 1
        else
            right -= 1
        end
    end
    return max_area
end


@testset "maxArea tests" begin
    # Test case 1
    height = [1, 8, 6, 2, 5, 4, 8, 3, 7]
    expected_output = 49
    @test maxArea(height) == expected_output
    
    # Test case 2
    height = [1, 1]
    expected_output = 1
    @test maxArea(height) == expected_output
    
    # Test case 3
    height = [4, 3, 2, 1, 4]
    expected_output = 16
    @test maxArea(height) == expected_output
    
    # Test case 4
    height = [1, 2, 1]
    expected_output = 2
    @test maxArea(height) == expected_output
end

File: test/rust/tests/11.rs
fn max_area(height: Vec<i32>) -> i32 {
    let (mut left, mut right) = (0, height.len() - 1);
    let mut max_area = 0;

    while left < right {
        let h_l = height[left];
        let h_r = height[right];
        let area = (right - left) as i32 * std::cmp::min(h_l, h_r);
        max_area = std::cmp::max(max_area, area);

        if h_l < h_r {
            left += 1;
        } else {
            right -= 1;
        }
    }

    max_area
}

#[cfg(test)]
mod tests {
    use super::*;

    #[test]
    fn test_example1() {
        let height = vec![1, 8, 6, 2, 5, 4, 8, 3, 7];
        assert_eq!(max_area(height), 49);
    }

    #[test]
    fn test_example2() {
        let height = vec![1, 1];
        assert_eq!(max_area(height), 1);
    }
}
